lib/cartopy/crs.py

```diff
@@ -2006,3 +2066,28 @@ def epsg(code):
     """
     import cartopy._epsg
     return cartopy._epsg._EPSGProjection(code)
+
+
+def from_proj4(proj4_terms, globe=None, bounds=None):
+    from cartopy._proj4 import get_proj4_dict, _PROJ4Projection
+    proj4_dict = get_proj4_dict(proj4_terms)
+
+    proj_to_crs = {
+        'lcc': LambertConformal,
+        'stere': NorthPolarStereo,
+    }
+    proj = proj4_dict['proj']
+    crs_class = proj_to_crs.get(proj)
+
+    # special cases
+    if proj == 'stere' and float(proj4_dict['lat_0']) < 0:
+        crs_class = SouthPolarStereo
+
+    # couldn't find a known CRS class
+    if crs_class is None and bounds is None:
+        raise ValueError("'bounds' must be specified for unknown CRS "
+                         "'{}'".format(proj))
+    elif crs_class is None:
+        return _PROJ4Projection(proj4_dict, globe=globe, bounds=bounds)
```

Comment:
I'm 👎 for creating a projection that cartopy hasn't implemented. (i.e. returning a generic ``_PROJ4Projection`` instance)

Reply:
I think we're going to disagree on this, but in the end it is your call. You mentioned in the issue that cartopy doesn't follow PROJ.4 exactly (not a 1:1 mapping) because some scientific fields have existing expectations about how to describe a CRS. In a similar line of thinking I would argue that some fields use PROJ.4 to describe everything so limiting a user to projections that have a custom CRS class seems odd. There will be projections that are available that cartopy doesn't support, if all of the abstract methods are implemented shouldn't a user be able to use whatever projection they want? If not, why support EPSG at all?

I understand wanting to have a nice CRS class name and everything, but if a PROJ.4 generic fallback class is available and the functionality is the same then does it really matter? I guess it would effectively stop development on new CRS classes for cartopy though. `</rant>`

Reply:
I should also mentioned that I'm the maintainer of https://github.com/pytroll/satpy and https://github.com/pytroll/pyresample. The reason I wanted to add this functionality was to make it easier for our users to take our `AreaDefinition` (PROJ.4 projection + bounds + number of pixels) and create a cartopy plot from them.

Reply:
Also if this doesn't get added to cartopy I will likely just add it to pyresample.